src/frontend/js/types/Joanie.ts

```diff
@@ -26,9 +26,12 @@ export interface Organization {
 
 export interface Contract {
```

Comment:
You should check the existing Serializer as contract payload will not have this shape.

https://github.com/openfun/joanie/blob/dead80989865da9398d005cf80e62c43d58a690e/src/backend/joanie/core/serializers/client.py#L285-L299

Nested information (`product_title', `course_code` and `course_cover` will be provided through the order related to the contract`). Also I did not see any `learner_name`

The contract `title` will be provided through the related contract definition.

`sign_date` is called `signed_on`

Reply:
As i know, the endpoint isn't finalized / merged yet.
My point here was not to assume how it would be done in joanie but to list every field that our frontend use.

This interface can then be used as reference to develop joanie endpoint and then richie can be updated with the real interface that matche joanie's serializer.